### PR TITLE
Transition from number to bigint

### LINKED ISSUE #177: Number into BigInt
In the future, the currency we will be using will be based on WEI, which the minimum value is represented by 10^18. This means the Number data type from JS/TS will no be sufficient and will overflow. 

In our case, the maximum number we would use safely, would be represented by ```Number.MAX_SAFE_INTEGER``` which is: 9007199254740991, taking this into account, the number data type can't safely handle the representation of 1 token in WEI.

### Describe the feature you'd like

The funds, quota and anything related to currency to be represented as BigInt.

### Additional context

"Safe" in this context refers to the ability to represent integers exactly and to compare them correctly. For example, ```Number.MAX_SAFE_INTEGER + 1 === Number.MAX_SAFE_INTEGER + 2``` will evaluate to true, which is mathematically incorrect. 

Even though the ```Number.MAX_VALUE``` is bigger, its not accurate anymore, and any value bigger than it is represented as ```Infinity```

Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

As we internally use mHOPR, according to the contract it uses 18 decimal places. So WEI is the way to correctly handle these values as precise as possible.
Source: https://gnosisscan.io/token/0x66225de86cac02b32f34992eb3410f59de416698?a=0x1f5deee62c65577802e3b15c3f2e523b2b772458#code

### PR DESCRIPTION
**Pull request by ItsLuized**
Closes #177

### CODE REVIEW

File: apps/discovery-platform/src/funding-service-api/index.spec.ts [hunk 1]
```diff
@@ -307,13 +352,13 @@ describe("test funding service api class", function () {
...
       const fundingResponse = await fundingServiceApi.requestFunds({
-        amount: 5,
+        amount: BigInt(5),
         node,
       });
 
       const successfulGetRequestStatusBody: getRequestStatusResponse = {
         accessTokenHash: "hash",
-        amount: "5",
+        amount: BigInt("5").toString(),
```
Comment (diegoalzate) [inline apps/discovery-platform/src/funding-service-api/index.spec.ts:361]: I feel like this is just writing "5" but with extra steps

Comment (diegoalzate) [reply apps/discovery-platform/src/funding-service-api/index.spec.ts:361]: if we are trying to express that a big int can come here in form of a string then thats fine, just in some places it is "5" and in others it is BigInt("5").toString().

File: apps/discovery-platform/src/graph-api/index.ts [hunk 2]
```diff
@@ -6,6 +6,9 @@ import { createLogger } from "../utils";
 import * as constants from "../constants";
 
 const log = createLogger(["graph-api"]);
+// @ts-ignore
```
Comment (diegoalzate) [inline apps/discovery-platform/src/graph-api/index.ts:9]: I think we can make this part ts compliant

File: apps/funding-service/src/entry-server/middleware/index.ts [hunk 3]
```diff
@@ -34,11 +34,16 @@ export const tokenIsValid =
       return res.status(401).json({ errors: "Access Token expired" });
     }
 
+    if (requestFunds && req.body.amount === undefined) {
+      log.verbose("missing funding amount");
+      return res.status(400).json({ errors: "Missing funding amount" });
+    }
```
Comment (diegoalzate) [inline apps/funding-service/src/entry-server/middleware/index.ts:40]: amount being inside the body is part of the validation done in express server [here](https://github.com/Rpc-h/RPCh/blob/bccebb35436508d7b8e18164a431ec4dd8668a91/apps/funding-service/src/entry-server/index.ts#L52)

File: apps/funding-service/src/queue/index.spec.ts [hunk 4]
```diff
@@ -113,16 +115,14 @@ describe("test index.ts", function () {
     const queryRequest = await requestService.getRequest(createRequest.id);
     assert.equal(queryRequest?.status, "REJECTED-DURING-PROCESSING");
   });
-  it("should fail if signer does not have enough to fund request", async function () {
+  it.only("should fail if signer does not have enough to fund request", async function () {
```
Comment (diegoalzate) [inline apps/funding-service/src/queue/index.spec.ts:118]: remove this .only

**Review by diegoalzate — CHANGES_REQUESTED** [review]

File: apps/funding-service/src/db/index.spec.ts [hunk 5]
```diff
@@ -14,6 +14,17 @@ export class MockPgInstanceSingleton {
 
   private constructor() {
     let instance = newDb();
+    instance.public.interceptQueries((sql) => {
```
Comment (nionis) [inline apps/funding-service/src/db/index.spec.ts:17]: lets add a comment to what this does

**Review by nionis — APPROVED** [review]

File: apps/funding-service/src/db/index.spec.ts (hunk 5 above)
Comment (nionis) [reply apps/funding-service/src/db/index.spec.ts:17]: and if its used in many places, we should somehow make it re-usable so its not copy pasted

File: packages/common/src/fixtures.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@ import { utils } from "ethers";
 import * as crypto from "@rpch/crypto-bridge/nodejs";
 import Request from "./request";
 import Response from "./response";
+import { IMemoryDb } from "pg-mem";
```
Comment (nionis) [inline packages/common/src/fixtures.ts:6]:
```suggestion
import type { IMemoryDb } from "pg-mem";
```

**Review by nionis — APPROVED** [review]

**Review by nionis — APPROVED** [review]

**Pull request merged**